sol/stack_check.hpp

```diff
@@ -31,9 +31,9 @@
 #include <functional>
 #include <utility>
 #include <cmath>
-#ifdef SOL_CXX17_FEATURES
+#ifdef SOL_STD_VARIANT
```

Comment:
The SOL_CXX_17_FEATURES check should still be around this bllock: both should be present.